### DISCUSSION TITLE
How to modify the Buffer for a WorldObject in a for-loop?

### DISCUSSION THREAD

**Discussion by anderssonf** [pygfx/pygfx#667] [Q&A]
I would like to update plots in fastplotlib during simulations in a for loop. It seems like this requires modifying the Buffer for a WorldObject during the for-loop (e.g. in jupyter), see https://github.com/fastplotlib/fastplotlib/discussions/417#discussion-6310085

Can this be done?

**Comment by Korijn** [comment]
Just render a new frame inside the for loop?

See https://github.com/pygfx/wgpu-py/issues/471

**Reply by anderssonf** [reply]
Yes, it would be very useful to be able to view results while computing.

Thanks for the help!

**Reply by Korijn** [reply]
See https://github.com/pygfx/pygfx/pull/669

**Reply by kushalkolar** [reply]
Thanks Korijn, I also made a corresponding example in fpl, same exact concept https://github.com/fastplotlib/fastplotlib/blob/main/examples%2Fnotebooks%2Fmultiprocessing_zmq%2Fmultiprocessing_zmq_plot.ipynb
